### DISCUSSION TITLE
Better object references navigation?

### DISCUSSION THREAD

**Discussion by dunrix** [joaotavora/eglot#1172] [Ideas]
Hello,

is it just me but find a bit awkward current way of navigation over identifier references? I mean, if you want to quickly traverse within a current buffer.

Now, you have to open a new window with `xref-find-references`, which causes loss of focus from the  original one and related cursor position. Selection process within *\*xref\** buffer is in no way reflected, like current reference highlighted by corresponding line position.

I'm looking for an equivalent to `lsp-ui-peek-forward` & `lsp-ui-peek-backward` [lsp-ui](https://emacs-lsp.github.io/lsp-ui/#lsp-ui-peek), or `NextReference` & `PreviousReference` of [vim-lsc](https://github.com/natebosch/vim-lsc), or even `LspPeekReferences` made in [vim9 lsp](https://github.com/yegappan/lsp).

**Comment by narendraj9** [comment]
Have you already tried `follow-mode` in xref buffers and concluded that it is not sufficient?

**Reply by dunrix** [reply]
I've tried already but can't see how it can remedy described issue. Navigation selection remains not emphasized in the target buffer.

**Comment by terlar** [comment]
I think this could be implemented as a show function for xref. Currently you can set `xref-show-xrefs-function` to something like `consult-xref` (from [consult](https://github.com/minad/consult)) for example to get a different interface. I have been thinking about implementing something like "xref-quick-peek" (inspired by [xref-posframe](https://github.com/fmdkdd/dotfiles/blob/master/emacs/.emacs.d/elisp/xref-posframe.el) or [source-peek](https://github.com/iqbalansari/emacs-source-peek)).

Probably something like [quick-peek](https://github.com/cpitclaudel/quick-peek) could be used to implement the display part.
